GrandCentralBoard/Widgets/Harvest/BillableDates.swift

```diff
@@ -0,0 +1,42 @@
+//
+//  BillableDates.swift
+//  GrandCentralBoard
+//
+//  Created by Maciek Grzybowski on 03.06.2016.
+//  Copyright © 2016 Oktawian Chojnacki. All rights reserved.
+//
+
+import Foundation
+
+
+final class BillableDates {
+
+    let dates: [NSDate]
+
+    init(referenceDate: NSDate, numberOfPreviousDays: Int, includeWeekends: Bool?) {
+        let calendar = NSCalendar(calendarIdentifier: NSCalendarIdentifierGregorian)!
+        let sunday = 1
+        let saturday = 7
+
+        var previousDays: [NSDate] = []
+        var date = referenceDate
+
+        while previousDays.count < numberOfPreviousDays {
```

Comment:
Yeah, true. Forgot that you are removing some elements, so you don't know what the sequence should be.